static/src/javascripts/projects/common/modules/discussion/comment-box.js

```diff
@@ -0,0 +1,593 @@
+                );
+            }
+
+            if (this.options.replyTo) {
+                comment.replyTo = this.options.replyTo;
+            }
+
+            if (this.errors.length === 0) {
+                if (this.options.newCommenter && !this.options.hasUsername) {
+                    IdentityApi.updateUsername(
+                        this.getElem('onboarding-username-input').value
+                    ).then(updateUsernameSuccess, updateUsernameFailure);
+                } else {
+                    postCommentToDAPI();
+                }
+            }
+        };
+
+        if (!this.getUserData().emailVerified) {
+            // Cookie could be stale so lets refresh and check from the api
+            const createdDate = new Date(this.getUserData().accountCreatedDate);
+
+            if (createdDate > this.options.priorToVerificationDate) {
+                IdentityApi.getUserFromApiWithRefreshedCookie().then(
+                    response => {
+                        if (
+                            response.user.statusFields.userEmailValidated ===
+                            true
+                        ) {
+                            validEmailCommentSubmission();
+                        } else {
+                            this.invalidEmailError();
+                        }
+                    }
+                );
+            } else {
+                validEmailCommentSubmission();
+            }
+        } else {
+            validEmailCommentSubmission();
+        }
+    }
+
+    invalidEmailError(): void {
+        this.removeState('onboarding-visible');
+        this.error('EMAIL_NOT_VALIDATED');
+        ValidationEmail.init();
+    }
+
+    submitPostComment(e: Event): void {
+        e.preventDefault();
+        this.postComment();
+    }
+
+    ready(): void {
+        if (this.getDiscussionId() === null) {
+            throw new Error(
+                'CommentBox: You need to set the "data-discussion-key" on your element'
+            );
+        }
+
+        const commentBody = this.getElem('body');
+
+        this.setFormState();
+
+        if (this.options.newCommenter) {
+            bean.on(document.body, 'submit', [this.elem], (event: Event) =>
+                this.showOnboarding(event)
+            );
+            bean.on(
+                document.body,
+                'click',
+                this.getClass('onboarding-cancel'),
+                (event: Event) => this.hideOnboarding(event)
+            );
+        } else {
+            bean.on(document.body, 'submit', [this.elem], (event: Event) =>
+                this.submitPostComment(event)
+            );
+        }
+
+        bean.on(document.body, 'change keyup', [commentBody], () =>
+            this.setFormState()
+        );
+
+        bean.on(commentBody, 'focus', () => this.setExpanded());
+
+        this.on('change', '.d-comment-box__body', this.resetPreviewComment);
```

Comment:
Does `this.resetPreviewComment` need to be wrapped in an anonymous function to preserve the context binding? This is the only instance of an event handler being added in which this pattern is not applied